### DISCUSSION TITLE
Better management of Rugby version updates

### DISCUSSION THREAD

**Discussion by dcacenabes** [swiftyfinch/Rugby#386] [Q&A] [answered]
I have installed Rugby in my machine using `curl -Ls https://swiftyfinch.github.io/rugby/install.sh | bash`. I was trying to update the Rugby version in my machine by running `rugby update`.
The command was failing with:

```
⚑ Update
  ◑ [0s] Loading Releases
⛔️ Error: typeMismatch(Swift.Array<Any>, Swift.DecodingError.Context(codingPath: [], debugDescription: "Expected to decode Array<Any> but found a dictionary instead.", underlyingError: nil))
```

I used Proxyman to inspect the http traffic, and I noticed that Rugby does an unauthenticated request to the Github api:
https://api.github.com/repos/swiftyfinch/rugby/releases?per_page=30

Github is sending the following response with a status code 403 forbidden:
```
{
  "message": "API rate limit exceeded for XXX.XXX.XXX.XXX. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)",
  "documentation_url": "https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"
}
```

I have 2 suggestions:
1. Improve error handling for this case in the Rugby app so that it outputs a more user friendly message.
2. Consider different distribution mechanism that can not get throttled by Github.

**Comment by dcacenabes** [comment]
I created a PR improving this:
https://github.com/swiftyfinch/Rugby/pull/387

**Reply by swiftyfinch** [reply]
Thank you! I will add one more soon.

**Reply by swiftyfinch** [reply, marked as answer]
- #388

**Reply by swiftyfinch** [reply]
🚢 [v2.8.0](https://github.com/swiftyfinch/Rugby/releases/tag/2.8.0)
